linode/manager
# refactor: [M3-6571] - RQ-ify Events - Part 2: Event Handlers

## Description 📝
*See Part 1 here: #9046.*

This PR finishes off the RQ Events migration, moving all event handlers over from the old system (Redux middleware + rxjs observable) to a unified events "mega-hook".

The 2 hooks from the last PR are combined into a single hook `useEventsInfiniteQuery`, which handles both fetching events and allows for registering event handlers for new/updated events. It was necessary to combine the polling hook into this one because we use previously-fetched events to create a list of in-progress `eventIds` to poll for updates.

The event handlers themselves have been moved to two new hooks, `useAppEventHandlers` and `useToastNotifications`, from `<App />` where most of them used to live (a few others were scattered around the codebase -- they are all now in one place). I decided to move them out of `App` to reduce clutter in that file; moving them to hooks instead of components seemed appropriate because they don't render any content.

This PR also introduces a container, `withEventsInfiniteQuery` that allows our class-based components to use events.

## What exactly do Events do ❓
Events have a few different functions in Cloud Manager:

* Notify the user of ongoing/completed changes and updates to their account entities (Linodes, NodeBalancers, etc.)
    * These Event notifications are displayed in the Notification Menu and the Events Landing at /events.
    * Toast notifications appear when long running tasks complete (i.e., a Linode migration)
* Trigger event handlers that update locally cached data
    * E.g., when a Linode is updated in another tab, the Linodes cache is invalidated and Cloud Manager requests and displays the latest details without having to refresh the page
* Display the current status of an entity
    * E.g., when a Linode resize is in progress, this information, along with the progress will be displayed in the Linode details page

## Major Changes 🔄
- Creates one RQ hook, `useEventsInfiniteQuery`, for retrieving events and registering event handlers
- Consolidate all event handlers to two new hooks, `useAppEventHandlers` and `useToastNotifications`
- Delete the Events Redux store and all associated files (reduces, actions, helpers, etc.)
- Remove `rxjs` as a dependency

## How to test 🧪
1. Open two tabs of this version of the app.
2. Modify some entities in one tab and ensure those modifications are reflected in another tab (e.g., renaming/resizing/rebooting/deleting Linodes/NodeBalancers/Firewalls, etc.)
3. Perform an event that takes some time to complete and reports progress updates (e.g., resizing or migrating a Linode). Ensure both tabs receive and display updates as the event progresses and eventually completes.
4. Refresh in the middle of an in-progress event and ensure it continues to update until completion.
5. Perform some action that creates a toast notification (see `useToastNotifications.tsx` for examples of this, such as attaching a volume, resizing a disk, or migrating a Linode). Ensure this toast actually appears in both tabs.

Additional things to check:
  - Ensure all event handlers in `useAppEventHandlers` are registered and invoked as expected
  - Ensure all toast notifications in `useToastNotifications` work as expected
  - Ensure the events polling interval (via `resetPollingInterval`) is correctly reset when adding/updating/deleting/performing actions on entities
  - Ensure the NotificationMenu, EventsLanding and LinodeActivity components continue to display events as expected
  - Ensure unit and E2E tests pass
